library/packages/src/lib/y2packager/resolvable.rb

```diff
@@ -80,6 +80,17 @@ def initialize(hash)
       from_hash(hash)
     end
 
+    # Backward compatibility method to access resolvable like hash.
+    def [](key)
+      log.info "Calling [] with #{key}. Deprecated. Use method name directly"
+
+      public_send(key.to_sym)
+    # key not found, so return nil to be compatible
+    rescue NameError
+      log.info "attribute #{key} not defined, returning nil."
```

Comment:
I think this should be a warning, the list of the attributes is more or less fixed, using a missing attribute is really wrong.

Reply:
hmm, so using release_type is error there at https://github.com/yast/yast-registration/blob/c9b2889d13857ebba404e9db00cecf27cb58ffdb/src/lib/registration/registration.rb#L118 ?

Reply:
Hm,`release_type` has never been supported in [Pkg.ResolvableProperties](https://github.com/yast/yast-pkg-bindings/blob/master/src/Resolvable_Properties.cc), but there is [register_release](https://github.com/yast/yast-pkg-bindings/blob/master/src/Resolvable_Properties.cc#L385) value.

It seems that the SUSE Connect also uses that (with some additional magic): https://github.com/SUSE/connect/blob/7f743bc184a87b995ea2f02008cdb3b22387b839/lib/suse/connect/zypper/product.rb#L38

But AFAIK that value is `nil` in most (all?) cases so far, so that was not a problem.